### PR TITLE
ATLEDGE-229 CurieRTC and Time Library

### PR DESCRIPTION
**Pull request by bigdinotech**
Initial implementation of the CurieRTC library for the built-in RTC
Includes the Time library plus 3 new examples for the Arduino101

### CODE REVIEW

**Comment by bigdinotech** [comment]
@kmsywula @SidLeung Ready for code review

File: libraries/CurieRTC/examples/SetTime/SetTime.ino [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#include <Wire.h>
```
Comment (kmsywula) [inline libraries/CurieRTC/examples/SetTime/SetTime.ino]: Wire.h not necessary here.

**Comment by bigdinotech** [comment]
@calvinatintel @kmsywula I have addressed the minor issues found

**Comment by bbaltz505** [comment]
KW complains about  CurieRTC::set not returning a value
  bool CurieRTC::set(time_t t)

**Comment by kmsywula** [comment]
Klocwork is right. If you don't want it to return a value it should be 'void CurieRTC::set(time_t T)'.

**Comment by bigdinotech** [comment]
@bbaltz505 @kmsywula I have addressed the issues found by Klocwork

**Comment by kmsywula** [comment]
@calvinatintel I reviewed the code, ready for merge

**Pull request merged**